process_request/routines.py

```diff
@@ -250,6 +250,11 @@ def prepare_reading_room_request(self, items, request_data):
             "ScheduledDate": request_data.get("scheduledDate"),
             "SpecialRequest": request_data.get("questions"),
         }
+        if request_data.get("readingRoomID"):
+            reading_room_defaults["Location"] = request_data.get("readingRoomID")
+        if request_data.get("site"):
+            reading_room_defaults["Site"] = request_data.get("site")
+
```

Comment:
It might be cleaner to just add these values directly to the defaults and then filter out null values in `get_request_data` by returning something like `{k: v for k, v in data.items() if v}` in line 293.